pkg/controllers/resources/namespaces/syncer.go

```diff
@@ -96,7 +104,28 @@ func (s *namespaceSyncer) Sync(ctx *synccontext.SyncContext, event *synccontext.
 
 func (s *namespaceSyncer) SyncToVirtual(ctx *synccontext.SyncContext, event *synccontext.SyncToVirtualEvent[*corev1.Namespace]) (_ ctrl.Result, retErr error) {
 	// virtual object is not here anymore, so we delete
-	return patcher.DeleteHostObject(ctx, event.Host, nil, "virtual object was deleted")
+	if event.VirtualOld != nil || translate.ShouldDeleteHostObject(event.Host) {
+		return patcher.DeleteHostObject(ctx, event.Host, event.VirtualOld, "virtual object was deleted")
+	}
+
+	virtualName := ""
+	for vname, hname := range ctx.Config.Sync.ToHost.Namespaces.Mappings.ByName {
+		if hname == event.Host.Name {
```

Comment:
Isn't this a pattern, does it make sense to do a direct comparison here, what happens if I use "*" in my mapping

Reply:
yeah thx for catching this, clearly forgot to update logic here to work with new translator. Updated 👍